spec/requests/api/prosecution_cases_spec.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Api::ProsecutionCases', type: :request do
+  describe 'GET /api/prosecution_cases' do
+    around do |example|
+      VCR.use_cassette('search_prosecution_case/by_prosecution_case_reference_success') do
+        example.run
+      end
+    end
+
+    let(:valid_query_params) { { filter: { prosecution_case_reference: 'TFL12345' } } }
+    let(:schema) { File.read('schema/schema.json') }
+
```

Comment:
possible refactor
```
before do
  get api_prosecution_cases_path, params: valid_query_params
end

it 'successful' do
  expect(response).to have_http_status(200)
end

 it 'matches the given schema' do
   valid = JSON::Validator.validate!(schema, response.body, fragment: '#/definitions/prosecution_case/definitions/resource_collection', strict: true)
    expect(valid).to be_truthy
end
```

I have written a little [RSpec matcher to cover validating](https://github.com/ministryofjustice/laa-court-data-ui/compare/add-mock-adaptor-data#diff-b9bae26c3443ee588ca9448022dd04d9) if you like

Reply:
Thanks, I like this matcher! will pull it into this codebase.